include/mbedtls/ssl.h

```diff
@@ -3908,10 +3927,44 @@ int mbedtls_ssl_read( mbedtls_ssl_context *ssl, unsigned char *buf, size_t len )
  *                 query the active maximum fragment length.
  *
  * \note           Attempting to write 0 bytes will result in an empty TLS
- *                 application record being sent.
+ *                 application record being prepared.
  */
 int mbedtls_ssl_write( mbedtls_ssl_context *ssl, const unsigned char *buf, size_t len );
 
+/**
+ * \brief          Dispatch all previously queued application data
+ *
+ * \param ssl      The SSL context to use.
+ *
+ * \return         \c 0 if all data previously queued for dispatch via
+ *                 mbedtls_ssl_write() has been passed to the underlying
+ *                 transport.
+ * \return         #MBEDTLS_ERR_SSL_WANT_READ or #MBEDTLS_ERR_SSL_WANT_WRITE
+ *                 if the underlying transport was not available.
```

Comment:
This would benefit from an explanation of how `WANT_READ` can result from an attempt to write (repeat of the explanation given for `mbedtls_ssl_write`).